### ISSUE TITLE
Renaming media folder while using BlobStorage Provider for Media results in losing media items.

### ISSUE THREAD

**Issue by aderderian** [OrchardCMS/Orchard#7741]
After setting up a simple media hierarchy and putting some images in while using the Azure Blob Media Storage Provider I renamed a folder and this resulted in the media items not showing up anymore. They were lost in BLOB storage and the database showed the folder paths with "\" instead of "/" pathing. I tried to fix it manually but the data was lost and could not be recovered.

This was in the Orchard dev branch.

**Comment by aderderian** [comment]
Some more insights on this one after working with the client. It may be a separate issue, but I believe it to be related in the reproduction steps here. The user was renaming folders as well and after renaming a folder the content items underneath did not update their pathing. For example, media item /documents/Distributions/test.pdf was present and the user renamed the folder to "distributions" lower case. This updated the folder name but the pathing for the media items was not changed and resulted issues downstream in retrieving items, replacing items, deleting items.

**Comment by Piedone** [comment]
Can it be that an exception happened during the operation (which should also be logged)? Because if there is an uncaught exception after the storage folder rename happens then while the storage path will remain changed the DB changes (i.e. the path change for all affected Media items) will be rolled back, causing an inconsistency between the DB and storage.

Issue about preventing such problems: #4366.

**Comment by aderderian** [comment]
Potentially…It seemed to happen after a rename.

**Comment by barthamark** [comment]
What I successfully reproduced here is that if I rename the folder it will work except if the new folder name differs only in casing - the folder is created and the file is there but it is broken (contains only white spaces). It seems that the issue is in the Azure SDK (I updated the NuGet package and still the same). Also successfully reproduced it in a .NET Core Console Application. Created an issue on the Azure Storage SDK GitHub page: https://github.com/Azure/azure-storage-net/issues/892
